### PR TITLE
Add version info into gosysl binary

### LINKED ISSUE #286: sysl -version should print the version of the binary
sysl -version should print the version of the binary

Use go compilers build meta data to support -version

### PR DESCRIPTION
**Pull request by kasunfdo**
Fixes #286 .

Changes proposed in this pull request:
- Add version info into sysl binary so that `sysl --version` will show the binary version

@anz-bank/sysl-developers

### CODE REVIEW

File: sysl2/sysl/sysl.go [hunk 1]
```diff
@@ -13,12 +13,17 @@ import (
 	"gopkg.in/alecthomas/kingpin.v2"
 )
 
+// Version contains the sysl binary version
+// nolint:gochecknoglobals
```
Comment (anzdaddy) [inline sysl2/sysl/sysl.go]: No whitespace before `nolint`. This is explicitly recommend by golangci-lint. https://github.com/golangci/golangci-lint#nolint
```go
//nolint:gochecknoglobals
```

File: .travis.yml [hunk 2]
```diff
@@ -58,8 +58,8 @@ script:
 - gradle test -b test/java/build.gradle
 - go test -coverprofile=coverage.txt -covermode=atomic github.com/anz-bank/sysl/...
 - npm test --prefix sysl2/sysl/sysl_js
-- GOOS=darwin GOARCH=amd64 go build -o gosysl/gosysl-darwin github.com/anz-bank/sysl/sysl2/sysl
-- GOOS=linux GOARCH=amd64 go build -o gosysl/gosysl-linux github.com/anz-bank/sysl/sysl2/sysl
+- GOOS=darwin GOARCH=amd64 go build -o gosysl/gosysl-darwin -ldflags "-X main.Version=${TRAVIS_TAG:1}" github.com/anz-bank/sysl/sysl2/sysl
+- GOOS=linux GOARCH=amd64 go build -o gosysl/gosysl-linux -ldflags "-X main.Version=${TRAVIS_TAG:1}" github.com/anz-bank/sysl/sysl2/sysl
```
Comment (anzdaddy) [inline .travis.yml]: How is `TRAVIS_TAG` computed?

Comment (anzdaddy) [inline .travis.yml]: Defaulting to 1 doesn't look right. Better to use something like "unspecified".

Comment (anzdaddy) [inline .travis.yml]: Needs a test, maybe add one to `test-gosysl.sh`. Since you can't predict the version number, just check that `--version` works.

Comment (anzdaddy) [inline .travis.yml]: Would be good track additional info. This can be done later. #299

File: sysl2/sysl/sysl.go [hunk 3]
```diff
@@ -13,12 +13,17 @@ import (
...
+var Version = "undefined"
```
Comment (anzdaddy) [inline sysl2/sysl/sysl.go]: Prefer `"unspecified"`.

File: .travis.yml (hunk 2 above)
Comment (kasunfdo) [reply .travis.yml]: It is the same as git tag

Comment (kasunfdo) [reply .travis.yml]: It is not defaulting to 1. As git tag it defined like `v0.3.0`, just removing `v` from tag.

Comment (kasunfdo) [reply .travis.yml]: Agree. Although not included here, I have already added build date and commit id. Will address through #299.

File: sysl2/sysl/sysl.go (hunk 1 above)
Comment (kasunfdo) [reply sysl2/sysl/sysl.go]: IDE formats automatically by adding a space. 😀
Should turn that off

File: .travis.yml (hunk 2 above)
Comment (kasunfdo) [reply .travis.yml]: Done

File: sysl2/sysl/sysl.go (hunk 3 above)
Comment (kasunfdo) [reply sysl2/sysl/sysl.go]: Changed.

Comment (kasunfdo) [reply sysl2/sysl/sysl.go]: Changed

File: sysl2/sysl/sysl.go [hunk 4]
```diff
@@ -15,12 +15,17 @@ import (
 	"gopkg.in/alecthomas/kingpin.v2"
 )
 
+// Version contains the sysl binary version
+//nolint:gochecknoglobals
+var Version = "unspecified"
```
Comment (springwiz) [inline sysl2/sysl/sysl.go]: Version should be a const.

Comment (kasunfdo) [reply sysl2/sysl/sysl.go]: Although it sounds like a constant, cannot use constants as value should be reassigned during compile time

File: .travis.yml (hunk 2 above)
Comment (anzdaddy) [reply .travis.yml]: Ah, silly me. I mistook `:` for `:-`.

**Pull request merged**